ldelf/tlsdesc_rel_a64.S

```diff
@@ -0,0 +1,17 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2020, Huawei Technologies Co. Ltd.
+ */
+
+#include <asm.S>
+
+/*
+ * long tlsdesc_resolve(struct tlsdesc *);
+ *
+ * Must preserve all registers except x0, x1 and the processor flags.
```

Comment:
Do you have a reference to this?

Reply:
This sentence comes from https://www.fsfla.org/~lxoliva/writeups/TLS/RFC-TLSDESC-ARM.txt "Resolvers' Calling Convention" which is arm32 not arm64. However there is also this Android draft document which applies to several architectures: https://android.googlesource.com/platform/bionic/+/refs/tags/android-10.0.0_r45/docs/elf-tls.md#TLS-Descriptors-TLSDESC, it says "the resolver function has a special calling convention that preserves almost all registers".

I can try to find more details in the GCC code maybe.

Reply:
If you just add a comment with this it's good enough for me.